### DISCUSSION TITLE
How to make tab completion sorted by last modified?

### DISCUSSION THREAD

**Discussion by semi710** [junegunn/fzf#3793] [Q&A] [answered]
if doing tab completion to select a file?

**Comment by LangLangBart** [comment]
Is this working for you ?

```sh
source <(fzf --zsh)
_fzf_compgen_path() {
  rg --hidden --files --sortr=modified --glob '!.git/*' -- "$1"
}
```
> [!NOTE]
> Ensure the correct widget is assigned to <kbd>⇥ Tab</kbd>, rather than a different widget, such as from `Aloxaf/fzf-tab`.
>
> https://github.com/junegunn/fzf/blob/01e7668915c4e3cf8c9eeca283d41beac924fe1f/shell/completion.zsh#L365-L368
>
> ```sh
> bindkey | grep -F "^I"
> # Expected output: "^I" fzf-completion
>```

---

See also:

- [README - Settings](https://github.com/junegunn/fzf#settings)

Relevant code sections:

https://github.com/junegunn/fzf/blob/01e7668915c4e3cf8c9eeca283d41beac924fe1f/shell/completion.zsh#L189-L194

https://github.com/junegunn/fzf/blob/01e7668915c4e3cf8c9eeca283d41beac924fe1f/shell/completion.zsh#L162-L164

**Comment by semi710** [comment]
i got an sketchy solution for this https://github.com/niksingh710/cdots/blob/master/home/.shell/fzf.zsh#L100-L108

i was not aware of this path_completion and functions will look into them.
if you can help to improve my fzf config it will be helpful.

**Reply by LangLangBart** [reply, marked as answer]
> ```bash
> export FZF_CTRL_T_COMMAND="fd -H -tf --follow ${exclude_list[*]} | xargs ls -t"
> export FZF_ALT_C_COMMAND="fd $dirPATH -td ${exclude_list[*]} | xargs ls -dt"
> ```

Have a look at [Recommended way to sort results · Issue #196 · sharkdp/fd](https://github.com/sharkdp/fd/issues/196).
